### PR TITLE
Workflow and azure updates

### PR DESCRIPTION
**Pull request by mavaylon1**
This creates a workflow that will run a check on for building the docs on both nwb-schema and hdmf-common-schema. 

This also updates the azure_pipeline to use:
1. Latest versions on mac, linux, windows
2. Python support for 3.8 and 3.11
3. Removal of python 3.7

### CODE REVIEW

**Comment by mavaylon1** [comment]
@rly Why are we using an azure pipeline instead of a github actions to run a style check and pytest? I also updated the pipeline to use the latest versions for each operating system instead of being locked to a specific version.

**Comment by mavaylon1** [comment]
@oruebel This workflow will clone the repos for hdmf-common-schema and nwb-schema, checking that the docs can build correctly.

**Comment by mavaylon1** [comment]
Fix #78

**Comment by mavaylon1** [comment]
@rly @oruebel  There is no PR merge restrictions but give this a gander before a merge.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -8,13 +8,13 @@ jobs:
 - job: 'StyleCheck'
 
   pool:
-    vmImage: 'ubuntu-20.04'
+    vmImage: 'ubuntu-latest'
```
Comment (oruebel) [inline azure-pipelines.yml:11]: I thought we are using GitHub actions? Is this file needed?

Comment (mavaylon1) [reply azure-pipelines.yml:11]: @oruebel In general yes, but this repo does not have them setup. I will make a ticket for that, but this will get it updated till then. Talking with Ryan in our group chat, this was before the use of actions.
